pkgs/development/libraries/qt-6/modules/qtbase.nix

```diff
@@ -0,0 +1,451 @@
+    "-sctp"
+    "-libproxy"
+    "-sqlite" "system"
+
+    "-gui"
+    "-widgets"
+    "-opengl desktop"
+    "-libproxy"
+    "-sctp"
+    "-icu"
+    "-L" "${icu.out}/lib"
+    "-I" "${icu.dev}/include"
+    "-pch"
+  ] ++ lib.optional debugSymbols "-debug"
+    ++ lib.optionals developerBuild [
+    "-developer-build"
+    "-no-warnings-are-errors"
+  ] ++ (if (!stdenv.hostPlatform.isx86_64) then [
+    "-no-sse2"
+  ] else
+
+  [
+    "-sse2"
+    "${lib.optionalString (!stdenv.hostPlatform.sse3Support)   "-no"}-sse3"
+    "${lib.optionalString (!stdenv.hostPlatform.ssse3Support)  "-no"}-ssse3"
+    "${lib.optionalString (!stdenv.hostPlatform.sse4_1Support) "-no"}-sse4.1"
+    "${lib.optionalString (!stdenv.hostPlatform.sse4_2Support) "-no"}-sse4.2"
+    "${lib.optionalString (!stdenv.hostPlatform.avxSupport)    "-no"}-avx"
+    "${lib.optionalString (!stdenv.hostPlatform.avx2Support)   "-no"}-avx2"
+  ]
+  ) ++ [
+    "-no-mips_dsp"
+    "-no-mips_dspr2"
+  ] ++ [
+    "-system-zlib"
+    "-L" "${zlib.out}/lib"
+    "-I" "${zlib.dev}/include"
+    "-system-libjpeg"
+    "-L" "${libjpeg.out}/lib"
+    "-I" "${libjpeg.dev}/include"
+    "-system-harfbuzz"
+    "-L" "${harfbuzz.out}/lib"
+    "-I" "${harfbuzz.dev}/include"
+    "-system-pcre"
+    "-openssl-linked"
+    "-L" "${openssl.out}/lib"
+    "-I" "${openssl.dev}/include"
+    "-system-sqlite"
+    ''-${if libmysqlclient != null then "plugin" else "no"}-sql-mysql''
+    ''-${if postgresql != null then "plugin" else "no"}-sql-psql''
+
+    "-make libs"
+    "-make tools"
+    ''-${lib.optionalString (!buildExamples) "no"}make examples''
+    ''-${lib.optionalString (!buildTests) "no"}make tests''
+  ] ++ (
+    if stdenv.isDarwin then [
+      "-platform macx-clang"
+      "-no-fontconfig"
+      "-qt-freetype"
+      "-qt-libpng"
+      "-no-framework"
+    ] else [
+      "-xcb"
+      "-qpa xcb"
+      "-L" "${libX11.out}/lib"
+      "-I" "${libX11.out}/include"
+      "-L" "${libXext.out}/lib"
+      "-I" "${libXext.out}/include"
+      "-L" "${libXrender.out}/lib"
+      "-I" "${libXrender.out}/include"
+
+      "-libinput"
+
+      ''-${lib.optionalString (cups == null) "no-"}cups''
+      "-dbus-linked"
+      "-glib"
+      "-system-libpng"
+    ] ++ lib.optional withGtk3 "-gtk"
+      ++ [
+      "-inotify"
+      # Without these, Qt stops working on kernels < 3.17. See:
+      # https://github.com/NixOS/nixpkgs/issues/38832
+      "-no-feature-renameat2"
+      "-no-feature-getentropy"
+    ] ++ lib.optionals (cups != null) [
+      "-L" "${cups.lib}/lib"
+      "-I" "${cups.dev}/include"
+    ] ++ lib.optionals (libmysqlclient != null) [
+      "-L" "${libmysqlclient}/lib"
+      "-I" "${libmysqlclient}/include"
+    ]
+  );
+
+  # Move selected outputs.
+  postInstall = ''
+    set -o xtrace # debug
+
+    # debug
+    echo "postInstall: pwd = $(pwd)"
+    echo "postInstall: find mkspecs"
+    find mkspecs
+    echo "postInstall: find build/mkspecs"
+    find build/mkspecs
```

Comment:
With respect to the linked issue, the correct way to find the interpreter in a Nix build is `cat $NIX_CC/nix-support/dynamic-linker`. Is it worth it to upstream such a trivial patch, versus using `substituteInPlace`, when the solution doesn't generalize to other platforms anyway?

Reply:
> the correct way to find the interpreter in a Nix build is `cat $NIX_CC/nix-support/dynamic-linker`

you mean this patch?

```diff
-  readelf -l /bin/ls
+  readelf -l $(which ls)

-  /bin/pwd
+  pwd
```

how is `ld-linux.so` relevant here?

> Is it worth it to upstream such a trivial patch

yepp, cos its an upstream bug : )
calls like `/bin/pwd`, `/bin/ls`, `/bin/rm` are not portable, and not needed in shell scripts
but no worry, thats not the reason why my draft is moving slow ...

currently, im splitting buildPhase and installPhase into 2 derivations
since buildPhase takes 2 hours (on my machine) and installPhase takes like 2 minutes ...
so debugging the installPhase is a terrible feedback loop
i hope i can generalize this build/install splitting for other large builds.
once the installPhase is working, the splitting can be disabled

with the `ninja` build tool, this is a challenge,
cos ninja validates files and paths before install (on failure calls `rerun_cmake`),
and i have to patch file times (and maybe checksums) ...
this should be easier with `cmake` (but its slower than ninja, so they say)

Reply:
> currently, im splitting buildPhase and installPhase into 2 derivations

done in 3e50ae7 : )